src/main/java/duke/storage/Storage.java

```diff
@@ -0,0 +1,131 @@
+package duke.storage;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import duke.exception.DukeException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.TaskList;
+import duke.task.ToDo;
+
+/**
+ * The storage for Duke application.
+ *
+ * @author Farrel Dwireswara Salim.
+ */
+public class Storage {
+    private final String filePath;
+
+    /**
+     * Constructs a new Storage instance.
+     *
+     * @param directoryName the name of the directory where storage is located.
+     * @param fileName the name of the file to store the tasks.
+     * @throws DukeException If the storage fails to be created.
+     */
+    public Storage(String directoryName, String fileName) throws DukeException {
+        this.filePath = directoryName + "/" + fileName;
+
+        try {
+            File storageDirectory = new File(directoryName);
+
+            if (!storageDirectory.exists()) {
+                storageDirectory.mkdir();
+            }
+
+            File storageFile = new File(this.filePath);
+
+            if (!storageFile.exists()) {
+                storageFile.createNewFile();
+            }
+        } catch (IOException error) {
+            throw new DukeException(String.format("Failed to load tasks, %s",
+                    error.getMessage()));
+        }
+    }
+
+    /**
+     * Transforms the txt file in storage to its corresponding TaskList.
+     *
+     * @return the resulting TaskList, else if the storage fails to transform the txt file,
+     *     return an empty TaskList.
+     */
+    public TaskList loadTasksInStorage() {
+        TaskList loadedTaskList;
+
+        try {
+            List<Task> loadedTasks = new ArrayList<>();
+            BufferedReader storageReader = new BufferedReader(new FileReader(this.filePath));
+            String taskString = storageReader.readLine();
+
+            while (taskString != null && !taskString.isEmpty()) {
+                int dateSeparatorIndex = taskString.lastIndexOf("|");
+                int tagsIndex = taskString.indexOf("#");
+
+                String taskType = taskString.substring(0, 1);
```

Comment:
There might be some potential to execute SLAP here, and make parsing of the string another function's job.

Reply:
I believe using SLAP here can also help make the code here much shorter and readable.